### DISCUSSION TITLE
Best way to manage environment variables?

### DISCUSSION THREAD

**Discussion by keinsell** [vercel/turborepo#1558] [Help] [answered]
Hi, I was looking across the web (and repo as well) ways to actually manage `.env` across whole monorepo. Currently in my project I'm using small package based on `envalid` but I don't think it's a proper way to manage environent. How are you dealing with that, what are your ideas to do so?

```ts
import { config } from "dotenv";
import findUp from "find-up";
import { cleanEnv, str } from "envalid";

config({ path: findUp.sync(".env") });

const env = cleanEnv(process.env, {
  NODE_ENV: str({
    choices: ["development", "production"],
    default: "development",
  }),
});

export default env;
```

**Comment by anthonyshew** [comment, marked as answer]
We have updated our documentation for environment variable best practices here: https://turbo.build/repo/docs/crafting-your-repository/using-environment-variables#best-practices
